### ISSUE TITLE
Another bug with IPAdaper weights (if image batch is odd)

### ISSUE THREAD

**Issue by memo** [cubiq/ComfyUI_IPAdapter_plus#723]
If there are an odd number of images in the batch, it seems `weights` and `weights_invert` return the same list!
(zoom into the images below to see the numbers)

3 images in batch:
![image](https://github.com/user-attachments/assets/e02799a8-c9d7-4fe2-a2af-d366a1a749cc)

4 images in batch:
![image](https://github.com/user-attachments/assets/74926f8f-e93c-4947-a25e-621bfad2b1c5)

**Comment by jx1k** [comment]
I noticed the same. This happes, because `weights_invert` is the reversed `weights` list instead of the inverse (1.0 - weight) of each weight. With an odd number the values end on 1.0 and thus start on 1.0 when reversed.

To my understanding `weights_invert` is intended to drive a second IPAdapter that uses the image_2 output, so that two IPAdapter weights basically "crossfade". Maybe that's not what Matteo had in mind, but there is an easy fix to make it work in this way.

In IPAdapterPlus.py change line 1750 from

`weights_invert = weights[::-1]`

to 

`weights_invert = [1.0 - w for w in weights]`

**Comment by memo** [comment]
Yes, is this still happening? I'd filed this bug months ago and had to fix it locally and didn't check back. See also https://github.com/cubiq/ComfyUI_IPAdapter_plus/issues/723

**Comment by jx1k** [comment]
It's still in there. The code hasn't been updated since four month ago.
